src/script_lua.c

```diff
@@ -826,15 +847,45 @@ static robj **luaArgsToRedisArgv(lua_State *lua, int *argc) {
     if (j != *argc) {
         j--;
         while (j >= 0) {
-            decrRefCount(argv[j]);
+            decrRefCount(lua_argv[j]);
```

Comment:
i don't understand this change (or that hole code block actually).
i suppose it's because of the `break` in the loop above.
but i don't understand this change (and removal of `zfree(argv);` below.
isn't what's needed instead is to call freeLuaRedisArgv (maybe with `j` as an argument so it won't run on the full argv)?